### PR TITLE
Simplify dependency graph

### PR DESCRIPTION
**Pull request by fbiville**

### CODE REVIEW

**Review by ali-ince — APPROVED** [review]
looks good

File: src/main/java/org/neo4j/importer/v1/validation/plugin/NoDependencyCycleValidator.java [hunk 1]
```diff
@@ -95,19 +92,10 @@ public boolean report(Builder builder) {
...
-    private void trackDependency(Action action, String path) {
-        String targetName = action.getName();
-        namedPaths.put(targetName, path);
-        String dependencyName = action.getDependsOn();
-        if (dependencyName != null) {
-            dependencies.put(new Element(targetName, path), dependencyName);
-        }
+        Element element = new Element(targetName, path);
+        target.getDependencies().forEach((dependency) -> {
+            dependencies.put(element, dependency);
```
Comment (fbiville) [inline src/main/java/org/neo4j/importer/v1/validation/plugin/NoDependencyCycleValidator.java:97]: oops, this is wrong, fixing now

File: src/main/java/org/neo4j/importer/v1/validation/plugin/NoDanglingDependsOnValidator.java [hunk 2]
```diff
@@ -73,25 +67,15 @@ public boolean report(Builder builder) {
...
-    }
-
-    private void track(Action action, String path) {
-        names.add(action.getName());
-        String dependsOn = action.getDependsOn();
-        if (dependsOn != null) {
-            pathToDependsOn.put(path, dependsOn);
-        }
+        target.getDependencies().forEach(dependency -> {
+            pathToDependsOn.put(path, dependency);
```
Comment (fbiville) [inline src/main/java/org/neo4j/importer/v1/validation/plugin/NoDanglingDependsOnValidator.java:78]: this is also wrong, fixing now

File: src/main/java/org/neo4j/importer/v1/validation/plugin/NoDependencyCycleValidator.java (hunk 1 above)
Comment (fbiville) [reply src/main/java/org/neo4j/importer/v1/validation/plugin/NoDependencyCycleValidator.java]: fixed via https://github.com/neo4j/import-spec/pull/16/commits/76fdd7d3eb59d4bd0cf9e1f49a69420efe0d3dc5

File: src/main/java/org/neo4j/importer/v1/validation/plugin/NoDanglingDependsOnValidator.java (hunk 2 above)
Comment (fbiville) [reply src/main/java/org/neo4j/importer/v1/validation/plugin/NoDanglingDependsOnValidator.java]: fixed via https://github.com/neo4j/import-spec/pull/16/commits/76fdd7d3eb59d4bd0cf9e1f49a69420efe0d3dc5

File: src/test/java/org/neo4j/importer/v1/ImportSpecificationDeserializerExtraValidationTest.java [hunk 3]
```diff
@@ -937,4 +897,226 @@ void fails_if_all_targets_are_inactive() {
...
+                                .stripIndent())))
+                .isInstanceOf(InvalidSpecificationException.class)
+                .hasMessageContainingAll(
+                        "1 error(s)",
+                        "0 warning(s)",
+                        "$.targets.queries[0].depends_on defines dependency \"a-relationship-target\" 2 times, it must be defined at most once");
+    }
+
+    @Test
+    void does_not_report_if_targets_duplicate_dependency() {
```
Comment (fbiville) [inline src/test/java/org/neo4j/importer/v1/ImportSpecificationDeserializerExtraValidationTest.java:1066]:
```suggestion
    void does_not_report_cycles_if_targets_duplicate_dependency() {
```

**Pull request merged**